test/end2end_test.go

```diff
@@ -79,3 +79,38 @@ func TestEndToEnd(t *testing.T) {
 	}
 
 }
+
+*/
+func TestE2EUsingFetcherFlags(t *testing.T) {
+	// Set up
+	os.Create("specfile.txt")
+	err := os.WriteFile("specfile.txt", []byte("examples/csv_templates/template.csv"), 0644)
+	if err != nil {
+		t.Fatalf("Failed to create specfile.txt: %v", err)
+	}
+	CreateCsvFile()
+	defer CleanUp()
+	defer os.Remove("specfile.txt")
+
+	// Works on Unix and Windows
+	cmd := exec.Command("go", "build", "-o", "tech_radar.exe", "../src")
+	_, err = cmd.Output()
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	// Fetch files using CLI arguments and flags
+	cmd1 := exec.Command("./tech_radar.exe", "fetch", "https://github.com/NovoNordisk-OpenSource/decentralized-tech-radar", "--branch=main", "--whitelist=./specfile.txt")
+	_, err = cmd1.Output()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check if the file was downloaded
+	_, err = os.Stat("cache/template.csv")
+	if os.IsNotExist(err) {
+		t.Fatal("Failed to create Merged_file.csv")
+	}
+
+}
```

Comment:
I think I'm confused as to why this is a sub-E2E test rather than an integration test? o:

Reply:
It uses the binary directly. Integreation tests from what we were told in BDSA were just chained function calls but since this is a full on build and run test I decided to put it in there. Feel free to put it elsewhere if this is not the case